contest_runner.py

```diff
@@ -319,64 +319,79 @@ def upload_file(file_full_path, remote_name=None, remove_local=False):
 
-        # Process replays: compress and upload
-        replays_archive_name = "replays_%s.tar" % self.contest_timestamp_id
-        replays_archive_name += ".gz" if self.compress_logs else ""
-        replays_archive_full_path = os.path.join(
-            self.replays_archive_dir, replays_archive_name
-        )
-        with tarfile.open(
-            replays_archive_full_path, "w:gz" if self.compress_logs else "w"
-        ) as tar:
-            tar.add(self.tmp_replays_dir, arcname="/")
+        # PROCESS REPLAYS: compress and upload
+        replays_archive_name = 'replays_%s.tar' % self.contest_timestamp_id
+        replays_archive_name += '.gz' if self.compress_logs else ''
+        replays_archive_full_path = os.path.join(self.replays_archive_dir, replays_archive_name)
+        with tarfile.open(replays_archive_full_path, 'w:gz' if self.compress_logs else 'w') as tar:
+            tar.add(self.TMP_REPLAYS_DIR, arcname='/')
         if self.upload_replays:
             try:
-                replays_file_url = self.upload_file(
-                    replays_archive_full_path, remove_local=False
-                )
-                data_stats["url_replays"] = replays_file_url.decode()
+                replays_file_url = self.upload_file(replays_archive_full_path, remove_local=False)
+                data_stats['url_replays'] = replays_file_url.decode()
             except Exception as e:
-                replays_file_url = os.path.relpath(
-                    replays_archive_full_path, self.www_dir
-                )
+                replays_file_url = os.path.relpath(replays_archive_full_path, self.www_dir)
         else:
-            replays_file_url = os.path.relpath(
-                replays_archive_full_path, self.www_dir
-            )  # stats-archive/stats_xxx.json
-
-        # Process replays: compress and upload
-        logs_archive_name = "logs_%s.tar" % self.contest_timestamp_id
-        logs_archive_name += ".gz" if self.compress_logs else ""
+            replays_file_url = os.path.relpath(replays_archive_full_path, self.www_dir)  # stats-archive/stats_xxx.json
+
+        # Copy folder
+        replays_folder_name = 'replays_%s' % self.contest_timestamp_id
+        replays_archive_full_path = os.path.join(self.replays_archive_dir, replays_folder_name)
+        shutil.copytree(self.TMP_REPLAYS_DIR, replays_archive_full_path)
+
+        # Create archive for each team
+        for t in self.team_stats.keys():
+            replays_folder_name = 'replays_%s' % self.contest_timestamp_id
+            replays_archive_name = f'replays_{t}.tar'
+            replays_archive_name += '.gz' if self.compress_logs else ''
+            replays_archive_full_path = os.path.join(self.replays_archive_dir, replays_folder_name, replays_archive_name)
+            replays_folder_full_path = os.path.join(self.replays_archive_dir, replays_folder_name)
+            os.system(f'tar zcf {replays_archive_full_path} {replays_folder_full_path}/*{t}*')
```

Comment:
This should use the Python lib `tar.`as we use above (line 336). We should avoid having to rely on external package when possible, but we are already using `tar` anyways.

If there is a reason why not to use tar package, let's use the more modern `subprocess.run`

Reply:
It is not necessary to change it to `subprocess.` as we don;'t care about the output.

However, we may want to use python `tarfile` directly, instead of calling tar externally: 

https://docs.python.org/3/library/tarfile.html

Reply:
We leave it as external tool as `tar`  (written in C) will be way faster